roachtest: ruby-pg failed

roachtest.ruby-pg [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16695625?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RoachtestNightlyGceBazel/16695625?buildTab=artifacts#/ruby-pg) on release-24.1.4-rc @ [75d6ac620fb2d223454026e220a6999cedbba41c](https://github.com/cockroachdb/cockroach/commits/75d6ac620fb2d223454026e220a6999cedbba41c):


```
(orm_helpers.go:229).summarizeFailed: 
Tests run on Cockroach v24.1.4-dev-75d6ac620fb2d223454026e220a6999cedbba41c
Tests run against ruby-pg v1.4.6
898 Total Tests Run
713 tests passed
185 tests failed
0 tests skipped
4 tests ignored
0 tests passed unexpectedly
1 test failed unexpectedly
0 tests expected failed but skipped
0 tests expected failed but not run
---
--- FAIL: running with sync_* methods PG::Connection times out after connect_timeout seconds -  (unexpected)
For a full summary look at the ruby-pg artifacts 
An updated blocklist (rubyPGBlocklist) is available in the artifacts' ruby-pg log
test artifacts and logs in: /artifacts/ruby-pg/run_1
```

Parameters:
 - <code>ROACHTEST_arch=amd64</code>
 - <code>ROACHTEST_cloud=gce</code>
 - <code>ROACHTEST_coverageBuild=false</code>
 - <code>ROACHTEST_cpu=4</code>
 - <code>ROACHTEST_encrypted=false</code>
 - <code>ROACHTEST_metamorphicBuild=false</code>
 - <code>ROACHTEST_ssd=0</code>
<details><summary>Help</summary>
<p>


See: [roachtest README](https://github.com/cockroachdb/cockroach/blob/master/pkg/cmd/roachtest/README.md)



See: [How To Investigate \(internal\)](https://cockroachlabs.atlassian.net/l/c/SSSBr8c7)



See: [Grafana](https://go.crdb.dev/roachtest-grafana/teamcity-16695625/ruby-pg/1725086469892/1725087013977)

</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #128771 roachtest: ruby-pg failed [C-test-failure O-roachtest O-robot P-3 T-sql-foundations branch-release-24.1.3-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*ruby-pg.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41796

--- a/pkg/cmd/roachtest/tests/ruby_pg_blocklist.go
+++ b/pkg/cmd/roachtest/tests/ruby_pg_blocklist.go
@@ -200,4 +200,5 @@ var rubyPGIgnorelist = blocklist{
 	`running with sync_* methods PG::Connection OS thread support Connection.new shouldn't block a second thread`:                                                                  "flaky",
 	`running with sync_* methods PG::Connection handles server close while asynchronous connect`:                                                                                   "flaky",
 	`running with sync_* methods PG::Connection multinationalization support respect and convert character encoding of input strings should convert error string to #put_copy_end`: "flaky",
+	`running with sync_* methods PG::Connection times out after connect_timeout seconds`:                                                                                           "flaky",
 }

